src/networks.jl

```diff
@@ -14,6 +14,7 @@ struct RegisterNet
 end
 
 function RegisterNet(graph::SimpleGraph, registers, vertex_metadata, edge_metadata, directed_edge_metadata)
+    glcnt[] = 0 # set the global counter of `guid`s to zero whenever a new network is initialized
```

Comment:
someone might be modeling multiple networks. No need to set it to zero, to avoid clashes

Reply:
On the other hand, this seems pretty useful when it comes to tests... So maybe we should not have a global counter, rather a counter per network?

Reply:
Its straight forward to make the counter local to each network, but then taking in to account the previous comment about adding `id` and `time` automatically inside the `tag!`, the local `id` now would be a field of the network and we would need to pass the network to `tag!` in order to access it

Reply:
A new private field in the network makes sense, but having to pass around additional arguments is definitely not workable. The goal is not just to make something possible, equally important is for the interface to be something usable and logical.

Most of the objects that are being tagged carry a reference to the more centralized objects, e.g. the way `get_time_tracker` works. Maybe the best solution is to have `get_time_tracker` provide the object tracking the ids (e.g. by having a new simulation type).

Or you can just stick to a global non-zeroed counter, while fixing the rest of the things. That is fairly straightforward to solve separately.

Reply:
I think making `net` a field of our own `Simulation` type would be the best way about it then, as you suggested above. For now lets work with a the non-zeroed global counter, I'll open an issue for it